homeassistant/components/sensibo/config_flow.py

```diff
@@ -24,6 +26,52 @@ class SensiboConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
 
     VERSION = 2
 
+    entry: config_entries.ConfigEntry | None
+
+    async def async_step_reauth(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle re-authentication with Sensibo."""
+
+        self.entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Confirm re-authentication with Sensibo."""
+        errors: dict[str, str] = {}
+
+        if user_input:
+            api_key = user_input[CONF_API_KEY]
+            try:
+                await async_validate_api(self.hass, api_key)
+            except AuthenticationError:
+                errors["base"] = "invalid_auth"
+            except ConnectionError:
+                errors["base"] = "cannot_connect"
+            except NoDevicesError:
+                errors["base"] = "no_devices"
+            except NoUsernameError:
+                errors["base"] = "no_username"
+            else:
+                if self.entry:
```

Comment:
We shouldn't get here if the entry is `None`.

We should assert that `self.entry is not None` instead

Reply:
Correct but this was purely to make mypy happy. So asserting should be fine.